src/sparsezoo/objects/model_folder_objects.py

```diff
@@ -0,0 +1,442 @@
+        :return: path to the tar archive file created
+        """
+        if self.path is None:
+            raise ValueError(
+                "Attempting to zip the Directory object files, "
+                "but `self.path` is None. "
+                "Class object requires pointer to parent "
+                "folder directory to know where to save the tar archive file."
+            )
+        parent_path = pathlib.PurePath(self.path).parent
+        tar_file_path = os.path.join(parent_path, self.name + ".tar.gz")
+        with tarfile.open(tar_file_path, "w") as tar:
+            for file in self.files:
+                tar.add(file.path)
+        return tar_file_path
+
+    def unzip(self, path_to_extract: Optional[str] = None) -> bool:
+        """
+        If Directory is a tar archive, extract it to:
+         - parent directory of the Directory object (by default)
+         - specified path (if `tar_file_path` provided).
+
+        Note: `path_to_extract` is relative to the working dir!
+
+        :param path_to_extract: path where the tar archive file is extracted to
+        :return boolean flag; True if archive is properly extracted
+        """
+
+        if path_to_extract is None:
+            path_to_extract = "."
+
+        _, *extension = self.path.split(".")
+        if extension != ["tar", "gz"]:
+            raise ValueError(
+                "Attempting to extract tar archive, but the extension "
+                f"of the Directory object is {extension}, not `.tar.gz`"
+            )
+
+        tar = tarfile.open(self.path, "r")
+        tar.extractall(path=path_to_extract)
+        tar.close()
+        return True
+
+    def __len__(self):
+        return len(self.files)
+
+    def _infer_path_from_files(self) -> Union[str, None]:
+        # if Directory object describes a local directory, we may try to infer the path
+        # from the attributes of the files contained within
+        paths = [os.path.dirname(file.path) for file in self.files]
+
+        # assert that all the files have the same dirname.
+        if len(set(paths)) == 1:
+            return paths[0]
+        else:
+            return None
+
+
+class FrameworkFiles(Directory):
+    """
+    Object that represents a directory with framework files.
+
+    :param files: list of files contained within the FrameworkFiles
+    :param name: name of the FrameworkFiles
+    :param path: path of the FrameworkFiles
+    :param url: url of the FrameworkFiles
+    """
+
+    def __init__(
+        self,
+        files: List[File],
+        name: str,
+        path: Optional[str] = None,
+        url: Optional[str] = None,
+    ):
+
+        super().__init__(files=files, name=name, path=path, url=url)
+
+        self.nested_folders_names = ["checkpoint_(.*)/", "logs/"]
+
+    # TODO: Add support for model cards (pull integration from
+    #  model.md and specify which files to validate)
+    def validate(self, integration: Optional[str] = None) -> bool:
```

Comment:
I'm thinking we may want to split out validation for different integrations into subclasses that may be implemented and then can be registered in a dictionary or something similar

Reply:
I propose we add it as a new task for the next sprint. Let's not include this feature into the current review.